internal/dataplane/parser/translators/ingress_atc.go

```diff
@@ -238,3 +239,148 @@ func sniMatcherFromSNIs(snis []string) atc.Matcher {
 	}
 	return atc.Or(matchers...)
 }
+
+// calculateExpressionRoutePriority calculates the priority of the generated route.
+// It basically follows the calculating method used in Kong's traditional compatible router:
+//   - First, sort by number of fields specified (methods, hosts, headers, paths, snis)
+//   - Then, if both routes have hosts, the routes having only plain(non-wildcard) hosts has
+//     higher priority than routes having at least one wildcard host
+//   - Then, sort by numer of different headers (maximum header count = 255).
+//   - Then, paths with regex match has higher priority than prefix match.
+//   - Then, sort by regex_priority field (not supported in KIC with expression routes).
+//   - At last, sort by maximum length of paths in the route.
+func calculateExpressionRoutePriority(
+	paths []netv1.HTTPIngressPath,
+	regexPathPrefix string,
+	ingressHost string,
+	ingressAnnotations map[string]string,
+) int {
+	matchFields := 0
+	plainHost := false
+	headerCount := 0
+	maxPathLength := 0
+	hasRegexPath := false
```

Comment:
`calculateExpressionRoutePriority` looks too complex to me to be a single function. Could we extract at least two separate functions and call them in here? The first one could calculate all the properties (`matchFields`, `plainHost`, `headerCount`, `maxPathLength`, `hasRegexPath`) and pass it to the second one, which would encode them into an integer. 

I imagine something like:

```go
func calculateExpressionRoutePriority(
	paths []netv1.HTTPIngressPath,
	regexPathPrefix string,
	ingressHost string,
	ingressAnnotations map[string]string,
) int {
	priorityTraits := CalculateIngressRoutePriorityTraits(paths, regexPathPrefix, ingressHost, ingressAnnotations)
	return priorityTraits.Encode()
}

type IngressRoutePriorityTraits struct {
	MatchFields   int
	PlainHost     bool
	HeaderCount   int
	MaxPathLength int
	HasRegexPath  bool
}

func CalculateIngressRoutePriorityTraits(
	paths []netv1.HTTPIngressPath,
	regexPathPrefix string,
	ingressHost string,
	ingressAnnotations map[string]string,
) IngressRoutePriorityTraits {
  ...
}

func (t IngressRoutePriorityTraits) Encode() int {
	...
}
```
This way we could cover both `CalculateIngressRoutePriorityTraits` and `Encode` with a reasonable degree of UTs - I think it's worth the effort since there are quite a lot of possible cases here and it would be nice to see them documented in tests.

Reply:
Done.

Reply:
I think we're missing unit tests for `EncodeToPriority`.

Reply:
Added tests.